pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -151,3 +162,21 @@ type HTTPSDConfig struct {
 	// +optional
 	Authorization *v1.SafeAuthorization `json:"authorization,omitempty"`
 }
+
+// KubernetesSDConfig allow retrieving scrape targets from Kubernetes' REST API
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config
+// +k8s:openapi-gen=true
+type KubernetesSDConfig struct {
+	// Role is role of the service in Kubernetes
+	// Currently only node role is supported
+	//+kubebuilder:validation:Enum=node
```

Comment:
(nit) to comply with the Kubernetes API [conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants), constant values should be CamelCase.
```suggestion
	//+kubebuilder:validation:Enum=Node
```